packages/keyring/src/__tests__/Keyring.spec.ts

```diff
@@ -277,3 +278,55 @@ describe('reset', () => {
     expect(Object.keys(localStorage).filter((key) => key.startsWith('account:0x')).length).toEqual(0);
   });
 });
+
+describe('changePassword', () => {
+  const NEW_PASSWORD = 'valid-password';
+
+  it('should throw out error if wallet is not initialized', async () => {
+    const keyring = new Keyring();
+
+    await expect(keyring.changePassword(PASSWORD, NEW_PASSWORD)).rejects.toThrowError(
+      new CoongError(ErrorCode.KeyringNotInitialized),
+    );
+  });
+
+  it('should throw out error if currentPassword is not correct', async () => {
+    const keyring = await initializeNewKeyring();
+
+    await expect(keyring.changePassword('incorrect-password', NEW_PASSWORD)).rejects.toThrowError(
+      new CoongError(ErrorCode.PasswordIncorrect),
+    );
+  });
+
+  it('should initialize the rawMnemonic with new password', async () => {
+    const keyring = await initializeNewKeyring();
+
+    const initializeWallet: Mock = vi.fn();
+    keyring.initialize = initializeWallet;
+
+    await keyring.changePassword(PASSWORD, NEW_PASSWORD);
+
+    expect(initializeWallet).toHaveBeenCalledWith(MNEMONIC, NEW_PASSWORD);
+  });
+
+  it('should call `saveAccount` if currentPassword and newPassword is valid', async () => {
+    const keyring = await initializeNewKeyring();
+
+    await keyring.createNewAccount('account 0', PASSWORD);
+    await keyring.createNewAccount('account 1', PASSWORD);
+
+    const saveAccountSpy = vi.spyOn(InnerKeyring.prototype as any, 'saveAccount');
+    await keyring.changePassword(PASSWORD, NEW_PASSWORD);
+
+    expect(saveAccountSpy).toHaveBeenCalledTimes(2);
+  });
+
+  it('could use new password to unlock accounts after change password', async () => {
+    const keyring = await initializeNewKeyring();
+
+    await keyring.createNewAccount('test-account', PASSWORD);
+    await keyring.changePassword(PASSWORD, NEW_PASSWORD);
+
+    await expect(keyring.unlock(NEW_PASSWORD)).toBeTruthy();
```

Comment:
1. Since `unlock` is an `async` function, we should expect it to be resolved, not to be truthy here.
```suggestion
    await expect(keyring.unlock(NEW_PASSWORD)).resolves();
```

2. We should also check if calling `unlock` on an account keypair if it's passed or not.